llama_stack/apis/files/files.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import List, Optional, Protocol, runtime_checkable
+
+from pydantic import BaseModel
+
+from llama_stack.providers.utils.telemetry.trace_protocol import trace_protocol
+from llama_stack.schema_utils import json_schema_type, webmethod
+
+
+@json_schema_type
+class FileUploadResponse(BaseModel):
+    id: str
+    url: str
+
+
+@json_schema_type
+class FileResponse(BaseModel):
+    key: str
+    bucket: str
+    mime_type: str
+    url: str
+    bytes: int
+    created_at: int
+
+
+@json_schema_type
+class ListFileResponse(BaseModel):
+    data: List[FileResponse]
+
+
+@runtime_checkable
+@trace_protocol
+class Files(Protocol):
+    @webmethod(route="/files", method="POST")
+    async def create_upload_session(
+        self,
+        bucket: str,
```

Comment:
What is a bucket? Is it a container for files (blobs)? Who defines these?

Reply:
Bucket will be like a label under which we will have a unique key. It is simply a way to group keys under a common name. For example label can be: images, datasets, videos etc. 

Will update comments in the code.

Reply:
You can look at buckets as single level namespaces where you add your keys. Bucket name should be unique so if you upload two files under and specify the same bucket, those files will be listed when you run GET /files/{bucket}.

Reply:
@jaideepr97 As @vladimirivic noted above, I think the same rough semantics as S3. If we need recursive retrievals, we could add an API for it or a parameter to the current retrieval API. Would that work? I don't think we want to create a full filesystem API here because it does not match with what underlying cloud providers typically provide.